src/planner/constify_now.c

```diff
@@ -63,7 +64,22 @@ is_valid_now_expr(OpExpr *op, List *rtable)
 	Assert(var->varno <= list_length(rtable));
 	RangeTblEntry *rte = list_nth(rtable, var->varno - 1);
 
-	const Dimension *dim = get_hypertable_dimension(rte->relid);
+	/*
+	 * If this query on a view we might have a subquery here
+	 * and need to peek into the subquery range table to check
+	 * if the constraints are on a hypertable.
+	 */
+	if (rte->rtekind == RTE_SUBQUERY)
+	{
+		flags = CACHE_FLAG_MISSING_OK;
```

Comment:
Could you add a comment why the `CACHE_FLAG_NOCREATE` flag should not be set in this case?